### PR TITLE
Gui remodel

### PR DESCRIPTION
**Pull request by ChangUo79**
Hi guys, I have updated the selectCommand to display the photo of the person.

### CODE REVIEW

**Comment by shawnlsj97** [comment]
Hey Erwin, kindly fix the checkstyle errors so that your pull request may be merged!

**Comment by ChangUo79** [comment]
Give me 5 minutes to rectify. Thanks!

**Comment by ChangUo79** [comment]
> 
> 
> Hey Erwin, kindly fix the checkstyle errors so that your pull request may be merged!

Fixed the checkstyle. Ready to merge!

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -235,10 +235,81 @@ image::CommitActivityDiagram.png[]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
-
-_{Explain here how the data encryption feature will be implemented}_
+=== Select feature
```
Comment (hellodommy) [inline docs/DeveloperGuide.adoc:310]: Hi Erwin, I think modifications to the Developer Guide (or actually anything not related to Gui remodel) should not be done in this branch.

Can consider doing Developer Guide changes in another second branch instead, so they will not come under the Gui remodel pull request.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -235,10 +235,81 @@ image::CommitActivityDiagram.png[]
...
+This sub feature is similar to the other features such as `Name` and `Phone` which has a prefix `i/` followed by the
+file name (e.g. `i/default.png`). Prior to adding the photo of a person, the image file that is going to be used has to
+be in the `\src\main\resources\images` directory. `Photo` takes in the file name as a `String` and retrieves the
+photo to be added or edited from the directory that was mentioned previously.
+
+*File name restriction*
+The file name of the images is restricted to alphanumerics only and the file format that will be used is `.png`. This
+is done to keep the file name simple.
+
+==== Design Consideration
```
Comment (hellodommy) [inline docs/DeveloperGuide.adoc:377]: Good consideration.

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -161,12 +161,19 @@ Example :
 `find n/James Ho`
 
 === Selecting a player : `select`
-Selects a member which will display their details.
+Selects a member by index which will display their personal information.
```
Comment (hellodommy) [inline docs/UserGuide.adoc]: We might encounter a merge conflict as I edited the User Guide as well 😢.

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 4]
```diff
@@ -25,13 +26,15 @@
             + PREFIX_EMAIL + "EMAIL "
             + PREFIX_ADDRESS + "ADDRESS "
             + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_IMAGE + "IMAGE]\n"
```
Comment (hellodommy) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:29]: Will the image be compulsory when we add a new person now?

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 5]
```diff
@@ -42,6 +44,7 @@
             + "[" + PREFIX_EMAIL + "EMAIL] "
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
             + "[" + PREFIX_TAG + "TAG]...\n"
+            + "[" + PREFIX_IMAGE + "IMAGE]\n"
```
Comment (hellodommy) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:47]: How will the naming be done?

File: src/test/java/seedu/address/logic/commands/SelectCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+
+class SelectCommandTest {
```
Comment (hellodommy) [inline src/test/java/seedu/address/logic/commands/SelectCommandTest.java:21]: Good effort on the test classes!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 7]
```diff
@@ -139,4 +140,19 @@ public static Tag parseTag(String tag) throws ParseException {
...
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String photo} into an {@code Photo}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code photo} is invalid.
+     */
+    public static Photo parsePhoto(String imageFilePath) throws ParseException {
```
Comment (hellodommy) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:150]: Are there restrictions to the size of the photo?

**Review by hellodommy — COMMENTED** [review]
Many big changes in this pull request. Good job Erwin!

Just a summary of my comments:

-  Developer Guide changes should not be in this pull request

- I want to know more about the addition of the photo component - Are all members required to be linked to a photo now?

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (ChangUo79) [reply docs/DeveloperGuide.adoc:310]: I will take note of that

File: src/main/java/seedu/address/logic/commands/AddCommand.java (hunk 4 above)
Comment (ChangUo79) [reply src/main/java/seedu/address/logic/commands/AddCommand.java:29]: It is optional, similar to tags. If the user did not include an image, a default image will be added.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 7 above)
Comment (ChangUo79) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:150]: Currently there is no restrictions

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 5 above)
Comment (ChangUo79) [reply src/main/java/seedu/address/logic/commands/EditCommand.java:47]: do you mean the naming of the images?

**Pull request merged**